httpclient5/src/test/java/org/apache/hc/client5/http/entity/mime/TestMultipartEntityBuilder.java

```diff
@@ -118,6 +118,20 @@ public void testMultipartCustomContentTypeParameterOverrides() throws Exception
                 entity.getContentType());
     }
 
+    @Test
+    public void testMultipartCustomContentTypeUsingAddParameter() {
+        final MultipartEntityBuilder eb = MultipartEntityBuilder.create();
+        eb.setMimeSubtype("related");
+        eb.addParameter(new BasicNameValuePair("boundary", "yada-yada"));
+        eb.addParameter(new BasicNameValuePair("charset", "ascii"));
+        eb.addParameter(new BasicNameValuePair("my", "stuff"));
+        eb.buildEntity();
+        final MultipartFormEntity entity =  eb.buildEntity();
+        Assert.assertNotNull(entity);
+        Assert.assertEquals("multipart/related; boundary=yada-yada; charset=US-ASCII; my=stuff",
```

Comment:
Interesting, you have added `ascii`, but the value has been changed. Is there any special handling? This is maybe related to https://issues.apache.org/jira/browse/HTTPCLIENT-2159.

Reply:
yes, looks weird. I gonna take a look to the ticket HTTPCLIENT-2159.

Reply:
> Interesting, you have added `ascii`, but the value has been changed. 

@michael-o I think the charset value got correctly normalized to `US-ASCII`.

Reply:
True...